### DISCUSSION TITLE
Consider updating `CODEOWNERS` file

### DISCUSSION THREAD

**Discussion by t-hamano** [WordPress/gutenberg#49419] [General]
Related Slack chat: https://wordpress.slack.com/archives/C02QB2JS7/p1680101380833059

The [CODEOWNERS](https://github.com/WordPress/gutenberg/blob/trunk/.github/CODEOWNERS) file allows you to specify GitHub user accounts for certain directories, and if the file changes that PR is making are contained in those directories, a review request will be automatically sent.

It can be difficult to choose who to designate as reviewers, especially for those submitting a PR for the first time. And I personally think that the current CODEOWNERS files are away from people who actually update those files often.

If the file were updated to match the current operational situation, I think that even if reviewers were not designated, review requests would be sent to the appropriate people, making it easier for first-time contributors' PRs to be reviewed.

I would be very interested in hearing your opinion.

**Comment by getdave** [comment]
This was [raised in Core Editor chat today](https://wordpress.slack.com/archives/C02QB2JS7/p1680101380833059). Here's what I said:

> I recently updated bits of that file to ensure I was tagged into bits of code I regularly maintain or work on.
> 
> Perhaps other regular contributors could commit to doing the same.
> 
> That said we should avoid one person becoming seen as solely responsible for any given thing as everyone should feel able to step away and disconnect without worry.
>
> I personally add myself where I think I can add value or I want to keep an eye on changes. But I don’t consider myself the guardian of the feature.

**Comment by nerrad** [comment]
Here's what I added as a comment to the recent Core Editor chat:

> Related to CODEOWNERS: One downside to using CODEOWNERS this way (even though I like it and use it for this reason) is that the people who are CODEOWNERS:
> - must have write permissions to the repository
> - are automatically requested for reviews.
> 
> So even though the intent is not that they are "guardians" of that section of the code, to new contributors submitting code it can appear as though they are fulfilling that role.

**Comment by priethor** [comment]
> So even though the intent is not that they are "guardians" of that section of the code, to new contributors submitting code it can appear as though they are fulfilling that role

So much this. Based on my own past experience and feedback I hear, this can be especially confusing for new contributors who expect the codeowners to engage with the reviews. I think we need to be more adamant about keeping the CODEOWNERS file up to date; GitHub sets the implicit expectations that code owners are the guardians, and the file shouldn't be used as a notification system at the expense of making PR reviewing more unclear.

**Reply by t-hamano** [reply]
> Based on my own past experience and feedback I hear, this can be especially confusing for new contributors who expect the codeowners to engage with the reviews.

I agree. I've seen a few times new contributors comment to inactive code owners saying, "Hey John, can you review this PR?"

**Reply by priethor** [reply]
I reached out to some inactive contributors from the codeowners file; they will reassess which areas they regularly look over, if any, and get back to us to update the codeowners file.

**Comment by t-hamano** [comment]
Since this file has been updated by the following two PRs, we should be able to close this discussion now. cc @aduth 

- https://github.com/WordPress/gutenberg/pull/82262
- https://github.com/WordPress/gutenberg/pull/82292
